src/Simplex/Messaging/Agent.hs

```diff
@@ -1735,339 +1788,424 @@ processSMPTransmission c@AgentClient {smpClients, subQ} (tSess@(_, srv, _), v, s
+                        smpConfirmation senderKey e2ePubKey e2eEncryption_ encConnInfo phVer agentVersion >> ack
+                      (SMP.PHEmpty, AgentInvitation {connReq, connInfo}) ->
+                        smpInvitation connReq connInfo >> ack
+                      _ -> prohibited >> ack
+                  (Just e2eDh, Nothing) -> do
+                    decryptClientMessage e2eDh clientMsg >>= \case
+                      (SMP.PHEmpty, AgentRatchetKey {e2eEncryption}) ->
+                        newRatchetKey e2eEncryption >> ack
+                      (SMP.PHEmpty, AgentMsgEnvelope _ encAgentMsg) -> do
+                        -- primary queue is set as Active in helloMsg, below is to set additional queues Active
+                        let RcvQueue {primary, dbReplaceQueueId} = rq
+                        unless (status == Active) . withStore' c $ \db -> setRcvQueueStatus db rq Active
+                        case (conn, dbReplaceQueueId) of
+                          (DuplexConnection _ rqs _, Just replacedId) -> do
+                            when primary . withStore' c $ \db -> setRcvQueuePrimary db connId rq
+                            case find ((replacedId ==) . dbQId) rqs of
+                              Just rq'@RcvQueue {server, rcvId} -> do
+                                checkRQSwchStatus rq' RSSendingQUSE
+                                void $ withStore' c $ \db -> setRcvSwitchStatus db rq' $ Just RSReceivedMessage
+                                enqueueCommand c "" connId (Just server) $ AInternalCommand $ ICQDelete rcvId
+                              _ -> notify . ERR . AGENT $ A_QUEUE "replaced RcvQueue not found in connection"
+                          _ -> pure ()
+                        let encryptedMsgHash = C.sha256Hash encAgentMsg
+                        tryError (agentClientMsg encryptedMsgHash) >>= \case
+                          Right (Just (msgId, msgMeta, aMessage, rcPrev)) -> do
+                            resetRatchetSync
+                            case aMessage of
+                              HELLO -> helloMsg >> ackDel msgId
+                              REPLY cReq -> replyMsg cReq >> ackDel msgId
+                              -- note that there is no ACK sent for A_MSG, it is sent with agent's user ACK command
+                              A_MSG body -> do
+                                logServer "<--" c srv rId "MSG <MSG>"
+                                notify $ MSG msgMeta msgFlags body
+                              QCONT addr -> qDuplex "QCONT" $ \conn' -> continueSending addr conn' >> ackDel msgId
+                              QADD qs -> qDuplex "QADD" $ \conn' -> qAddMsg qs conn' >> ackDel msgId
+                              QKEY qs -> qDuplex "QKEY" $ \conn' -> qKeyMsg qs conn' >> ackDel msgId
+                              QUSE qs -> qDuplex "QUSE" $ \conn' -> qUseMsg qs conn' >> ackDel msgId
```

Comment:
better keep qDuplex as / where it was, and use another function in EREADY where ackDel is not needed...